src/lug/Window/Win32/WindowImplWin32.hpp

```diff
@@ -8,17 +10,40 @@ namespace priv {
     
 class WindowImpl {
 public:
-    WindowImpl() = default;
+    WindowImpl(Window *);
     ~WindowImpl() = default;
 
     WindowImpl(const WindowImpl&) = delete;
     WindowImpl(WindowImpl&&) = delete;
     WindowImpl& operator=(const WindowImpl&) = delete;
     WindowImpl& operator=(WindowImpl&&) = delete;
 
-    bool create(uint16_t width, uint16_t height, const std::string& title, Style style);
-    void close();
-    void processEvents(std::queue<lug::Window::Event>& events);
+    bool    create(const std::string& title, Style style);
+    void    close();
+    bool    pollEvent(lug::Window::Event&);
+
+private:
+    void    processWindowEvents(UINT message, WPARAM wParam, LPARAM lParam);
+    void    registerWindow();
+    void    setFullscreen();
```

Comment:
I think that the name of this function isn't appropriate, because "set" means that you can just set it to `true` or `false`.